Source/Spatializer/HoloLens2/CircularBuffer.cpp

```diff
@@ -60,13 +58,10 @@ void CircularBuffer::WriteSamples(float* sourceBuffer, uint32_t samplesToWrite)
 
 void CircularBuffer::WriteSamples(float* sourceBuffer, uint32_t samplesToWrite, uint32_t stride)
 {
-    // Figure out if we're about to buffer overrun
-    auto bufferedSamples = m_BufferedSamples;
-    if (bufferedSamples + samplesToWrite > m_BufferSize)
-    {
-        // We are about to overrun. Drop some samples to make room
-        DropSamples(samplesToWrite);
-    }
+    auto bufferedSamples = m_BufferedSamples.load();

```

Comment:
I don't know that atomicizing m_BufferedSamples actually helps here. For this specific case, it's possible for the reader to run immediately after you read the value. Since you're only reading, not modifying, you could probably get away without atomic.

Reply:
I do think the atomic update's important when it's being written (further down in this function and also in WriteSamples).

Reply:
F'ing github diffs... how do I see the whole file with the green and red?!? The "view file" is just like "you're on your own, buddy"

-kicks rocks-

Reply:
I found the "expand all" button. Yes, modifying needs atomic. Good change.